server/server.go

```diff
@@ -507,8 +507,17 @@ func (s *Server) configureAccounts() error {
 	if opts.SystemAccount != _EMPTY_ {
 		// Lock may be acquired in lookupAccount, so release to call lookupAccount.
 		s.mu.Unlock()
-		_, err := s.lookupAccount(opts.SystemAccount)
+		acc, err := s.lookupAccount(opts.SystemAccount)
 		s.mu.Lock()
+		if err == nil && s.sys != nil && acc != s.sys.account {
+			// sys.account.clients (including internal client)/respmap/etc... are transferred separately
+			s.sys.account = acc
+			s.addSystemAccountExports(acc)
+			// can't hold the lock as go routine reading it may be waiting for lock as well
```

Comment:
That's fine though, meaning that you can send to this channel under the server lock. The internalSendLoop will pick up the change when the server lock is released (if loop is blocked trying to grab the server lock).
Even the way you do it here (releasing the lock, sending, then reacquiring) does not guarantee that the internalSendLoop will have time to refresh in that interval.

Reply:
This is a blocking channel as I want the go routine to return from reload when the update is already started, not a lot earlier.
My understanding of blocking channels is that both go routines have to be using the channel in order to make progress. 
However I am holding a lock and so, if the other go routine is waiting for the lock  it will never get to the receiving part.

Reply:
You can create the channel buffered (size1): `make(chan struct{}, 1)` and you should send using a select:
```
select {
  case s.sys.resetCh <- struct{}{}:
  default:
}
```

Reply:
We discussed and ok, you at least guarantee that all sys messages using the old one have been sent.